### PR TITLE
UseExceptionHandler throws if ExceptionHandlingPath not set

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #400 

cc @Eilon

### CODE REVIEW

File: src/Microsoft.AspNetCore.Diagnostics/Resources.resx [hunk 1]
```diff
@@ -247,4 +247,7 @@
   <data name="RuntimeInfoPage_Environment" xml:space="preserve">
     <value>Environment:</value>
   </data>
+  <data name="ExceptionHandlerOptions_NotConfiguredCorrectly" xml:space="preserve">
+    <value>The exception handler middleware options 'ExceptionHandlingPath' and 'ExceptionHandler' are both null. Please set either both or at least the 'ExceptionHandlingPath' when using 'app.UseExceptionHandler()'.</value>
+  </data>
```
Comment (jbagga) [inline src/Microsoft.AspNetCore.Diagnostics/Resources.resx]: Perhaps too verbose?

Comment (Tratcher) [reply src/Microsoft.AspNetCore.Diagnostics/Resources.resx]: Agreed, you can combine those to say you must set either the path or the handler.

File: test/Microsoft.AspNetCore.Diagnostics.Tests/ExceptionHandlerTest.cs [hunk 2]
```diff
@@ -494,5 +494,26 @@ public class ExceptionHandlerTest
...
+            // Arrange
+            DiagnosticListener diagnosticListener = null;
+
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    diagnosticListener = app.ApplicationServices.GetRequiredService<DiagnosticListener>();
+
+                    // Act
+                    var exception = Assert.Throws<InvalidOperationException>(() => app.UseExceptionHandler());
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Diagnostics.Tests/ExceptionHandlerTest.cs]: The exception won't be thrown until you call builder.Build().

Comment (jbagga) [reply test/Microsoft.AspNetCore.Diagnostics.Tests/ExceptionHandlerTest.cs]: No, the exception is thrown from the ctor. Unless that isn't what should happen

**Pull request merged**